spec/travis/model/repository_provider/stash_provider_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+
+describe StashProvider do
+  include Support::ActiveRecord
+
+  let(:repo)    { Repository.new(owner_name: 'travis-ci', name: 'travis-ci') }
+  let(:provider) { StashProvider.new(repo) }
+  let(:content_params) { {project_key: 'travis-ci', repository_name: 'travis-ci', ref: '12345678', path: '.travis.yml'} }
+
+  it 'produces correct private cloning url' do
```

Comment:
it "use configured protocol for private repository"

Even better (and probably too complex is use `context` for defining